lib/spack/spack/fetch_strategy.py

```diff
@@ -292,7 +292,15 @@ def mirror_id(self):
 
     @property
     def candidate_urls(self):
-        return [self.url] + (self.mirrors or [])
+        urls = []
+
+        for url in [self.url] + (self.mirrors or []):
+            if url.startswith('file://'):
+                path = urllib_parse.quote(url.removeprefix('file://'))
```

Comment:
`removeprefix` is only available for Python >= 3.9. I think what you're doing is better than how I did it, but you might have to use the `re` library to get a solution that works for all Python versions that we want to support.

Reply:
Sorry, apparently I assumed `removeprefix` was so basic that it should have always been available. :smile: I went back to the `len` approach.